react/__tests__/SliderTrack.test.tsx

```diff
@@ -0,0 +1,272 @@
+    )
+
+    let renderedSlides = INDEXES.map(idx =>
+      queryByText(`I am a slide. My index is ${idx}`)
+    ).filter(Boolean)
+
+    expect(renderedSlides).toHaveLength(TOTAL_ITEMS)
+
+    // non-infinite slider
+    mockedUseSliderState.mockImplementationOnce(
+      () => mockInitialNonInfiniteSliderState
+    )
+
+    rerender(
+      <SliderTrack
+        usePagination={false}
+        totalItems={TOTAL_ITEMS}
+        infinite={false}
+      />
+    )
+
+    renderedSlides = INDEXES.map(idx =>
+      queryByText(`I am a slide. My index is ${idx}`)
+    ).filter(Boolean)
+
+    expect(renderedSlides).toHaveLength(TOTAL_ITEMS)
+  })
+})
+
+describe('Behavior upon interaction', () => {
+  it('should correct slider state after infinite loop transitions', () => {
+    const TOTAL_ITEMS = 10
+
+    // This is a state where the slider is in its first page clone
+    mockedUseSliderState.mockImplementation(() => ({
+      ...mockInitialInfiniteSliderState,
+      currentSlide: 10,
+      transform: mockInitialInfiniteSliderState.transformMap[10],
+    }))
+
+    const { getByTestId, rerender } = render(
+      <SliderTrack totalItems={TOTAL_ITEMS} infinite usePagination />
+    )
+
+    let renderedTrack = getByTestId('slider-track')
+
+    fireEvent.transitionEnd(renderedTrack)
+
+    expect(mockDispatch).toHaveBeenCalledWith({ type: 'DISABLE_TRANSITION' })
+    expect(mockDispatch).toHaveBeenCalledWith({
+      type: 'ADJUST_CURRENT_SLIDE',
+      payload: {
+        currentSlide: 0,
+        transform: mockInitialInfiniteSliderState.transformMap[0],
+      },
+    })
+
+    // This is a state where the slider is in its last page clone
+    mockedUseSliderState.mockImplementation(() => ({
+      ...mockInitialInfiniteSliderState,
+      currentSlide: -1,
+      transform: mockInitialInfiniteSliderState.transformMap[-1],
+    }))
+
+    rerender(<SliderTrack totalItems={TOTAL_ITEMS} infinite usePagination />)
+
+    renderedTrack = getByTestId('slider-track')
+    fireEvent.transitionEnd(renderedTrack)
+
+    expect(mockDispatch).toHaveBeenCalledWith({ type: 'DISABLE_TRANSITION' })
+    expect(mockDispatch).toHaveBeenCalledWith({
+      type: 'ADJUST_CURRENT_SLIDE',
+      payload: {
+        currentSlide: 5,
+        transform: mockInitialInfiniteSliderState.transformMap[5],
+      },
+    })
+  })
+
+  mockedUseSliderState.mockClear()
+})
+
+describe('Accessibility', () => {
+  it('should have correct aria attributes set for each slide', () => {
+    mockedUseSliderState.mockImplementation(
+      () => mockInitialInfiniteSliderState
+    )
+
+    const TOTAL_ITEMS = 10
+    const SLIDES_PER_PAGE = 5
+    const INDEXES = Array.from(new Array(10), (_, i) => i + 1)
+
+    const { getByTestId, getByLabelText } = render(
+      <SliderTrack usePagination totalItems={TOTAL_ITEMS} infinite />
+    )
+
+    const renderedSliderTrack = getByTestId('slider-track')
+
+    const clones = renderedSliderTrack.querySelectorAll(
```

Comment:
[You could use this](https://testing-library.com/docs/dom-testing-library/api-queries#byrole)

Reply:
This selector doesn't work in this case, not sure why... maybe it doesn't handle well elements with fallback roles, such as "none presentation". I've tried querying for `none` and for `presentation`, and both would return an empty array.